tests/test_modeling_nystromformer.py

```diff
@@ -0,0 +1,343 @@
+        self.model_tester.create_and_check_for_question_answering(*config_and_inputs)
+
+    def test_for_sequence_classification(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_sequence_classification(*config_and_inputs)
+
+    def test_for_token_classification(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_token_classification(*config_and_inputs)
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in NYSTROMFORMER_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = NystromformerModel.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+    @unittest.skip(reason="Head masking not implemented")
+    def test_head_pruning(self):
+        pass
+
+    @unittest.skip(reason="Head masking not implemented")
+    def test_head_pruning_integration(self):
+        pass
+
+    @unittest.skip(reason="Head masking not implemented")
+    def test_head_pruning_save_load_from_config_init(self):
+        pass
+
+    @unittest.skip(reason="Head masking not implemented")
+    def test_head_pruning_save_load_from_pretrained(self):
+        pass
+
+    @unittest.skip(reason="Head masking not implemented")
+    def test_headmasking(self):
+        pass
+
+@require_torch
+class NystromformerModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_no_head(self):
+        model = NystromformerModel.from_pretrained("uw-madison/nystromformer-512")
+        input_ids = torch.tensor([[0, 1, 2, 3, 4, 5]])
+        output = model(input_ids)[0]
+
+        expected_shape = torch.Size((1, 6, 768))
+        self.assertEqual(output.shape, expected_shape)
+
+        # TODO Replace values below with what was printed above.
+        expected_slice = torch.tensor(
+            [[[-0.4532, -0.0936, 0.5137], [-0.2676, 0.0628, 0.6186], [-0.3629, -0.1726, 0.4716]]]
+        )
+
+        self.assertTrue(torch.allclose(output[:, :3, :3], expected_slice, atol=1e-4))
+
+    @slow
+    def test_inference_masked_lm(self):
+        model = NystromformerForMaskedLM.from_pretrained("uw-madison/nystromformer-512")
+        input_ids = torch.tensor([[0, 1, 2, 3, 4, 5]])
+        output = model(input_ids)[0]
+        vocab_size = 30000
+
+        expected_shape = torch.Size((1, 6, vocab_size))
+        self.assertEqual(output.shape, expected_shape)
+
+        # TODO Replace values below with what was printed above.
+        expected_slice = torch.tensor(
+            [[[-4.2054, 2.5545, -4.3157], [-7.1428, 23.0433, -4.2694], [-5.4785, 4.9240, -2.6824]]]
+        )
+
+        self.assertTrue(torch.allclose(output[:, :3, :3], expected_slice, atol=1e-4))
+
+    @slow
+    def test_masked_lm_end_to_end(self):
+        sentence = "the [MASK] of Belgium is Brussels"
+
+        tokenizer = AlbertTokenizer.from_pretrained("uw-madison/nystromformer-512")
```

Comment:
I would consider using `AutoTokenizer` here (to verify the `tokenizer_class` attribute is used properly).